test/ClientTest.php

```diff
@@ -524,4 +524,59 @@ public function testFormUrlEncodeSeparator()
         $rawRequest = $client->getLastRawRequest();
         $this->assertContains('foo=bar&baz=foo', $rawRequest);
     }
+
+    public function uriDataProvider()
+    {
+        return [
+            ['/example', true],
+            ['http://localhost/example', false]
+        ];
+    }
+
+    /**
+     * @dataProvider uriDataProvider
+     */
+    public function testValidRelativeURI($uri, $isValidRelativeURI)
+    {
+        $client = new Client($uri);
+        $this->assertSame($isValidRelativeURI, $client->getUri()->isValidRelative());
+
+        $client->setAdapter(Test::class);
+        $client->send();
+        $this->assertSame($isValidRelativeURI, $client->getUri()->isValidRelative());
+    }
+
+    public function portChangeDataProvider()
+    {
+        return [
+            ['https://localhost/example', 443],
+            ['http://localhost/example', 80]
+        ];
+    }
+
+    /**
+     * @dataProvider portChangeDataProvider
+     */
+    public function testAbsoluteSetPortWhenNoPort($absoluteURI, $port)
```

Comment:
This needs a name that better describes the behavior being tested: "testUriPortIsSetToAppropriateDefaultValueWhenAnAbsoluteUriOmittingThePortIsProvided".